src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaTests.cs

```diff
@@ -0,0 +1,83 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Security.Cryptography.SLHDsa.Tests
+{
+    [ConditionalClass(typeof(SlhDsaTestsBase), nameof(SupportedOnPlatform))]
```

Comment:
This condition is _slightly_ tricky.  SlhDsaKeyVault could work on a system that SlhDsa.GenerateKey does not.

I guess it means tests that test the relationship of the base type to the derived types (the boundary cases of the template method pattern) just need to be in a different class (which doesn't share the condition)